### PR TITLE
add documentation describing usage of test-record-size-limit.py

### PR DESCRIPTION
**Pull request by tomato42**
### Description

document the use of `test-record-size-limit.py` script

### Motivation and Context

The use of `test-record-size-limit.py` script is much harder and not typical for other scripts, add documentation that explains how it should be used.

### Checklist

- [x] I have read the [CONTRIBUTING.md](https://github.com/tomato42/tlsfuzzer/blob/master/CONTRIBUTING.md) document and my PR follows [change requirements](https://github.com/tomato42/tlsfuzzer/blob/master/CONTRIBUTING.md#change-requirements) therein
- [x] the changes are also reflected in documentation and code comments
- [x] all new and existing tests pass (see Travis CI results)
- [x] [test script checklist](https://github.com/tomato42/tlsfuzzer/wiki/Test-script-checklist) was followed for new scripts
- [x] new test script added to `tlslite-ng.json` and `tlslite-ng-random-subset.json`
- [x] new and modified scripts were ran against popular TLS implementations:
  - no code changes
- [x] required version of tlslite-ng updated in requirements.txt and README.md

---
This change is [Reviewable](https://reviewable.io/reviews/tomato42/tlsfuzzer/500)

### CODE REVIEW

File: USAGE.md [hunk 1]
```diff
@@ -417,8 +417,50 @@ and [test-truncating-of-kRSA-client-key-exchange.py](https://github.com/tomato42
 perform checks related to RSA key exchange. Failures there may be a sign of
 other problems.
 
+`test-record-size-limit.py`
+---------------------------
+
+This test is very stict on the size of returned reply and expects the reply
+to be sent in a single record, if its size allows for that.
+
+To quickly check what is the size that the server sends, it's possible to run
```
Comment (simo5) [inline USAGE.md]: remove that
as such -> with these options

File: USAGE.md [hunk 2]
```diff
@@ -417,8 +417,50 @@ and [test-truncating-of-kRSA-client-key-exchange.py](https://github.com/tomato42
...
+the test as such:
+```
+test-record-size-limit.py --reply-AD-size 1 \
+'check if server accepts maximum size in TLS 1.2'
+```
+or as such:
```
Comment (simo5) [inline USAGE.md]: remove "as such"

File: USAGE.md [hunk 3]
```diff
@@ -417,8 +417,50 @@ and [test-truncating-of-kRSA-client-key-exchange.py](https://github.com/tomato42
...
+```
+test-record-size-limit.py --reply-AD-size 1 \
+'check if server accepts maximum size in TLS 1.3'
+```
+
+It should fail with a
```
Comment (simo5) [inline USAGE.md]: with a -> with:

File: USAGE.md [hunk 4]
```diff
@@ -417,8 +417,50 @@ and [test-truncating-of-kRSA-client-key-exchange.py](https://github.com/tomato42
...
+`AssertionError: ApplicationData of unexpected size: X, expected: 1`
+where X is the value that needs to be passed as argument to `--reply-AD-size`.
+
+To make the test case more deterministic, it is possible to specify the
+exact (HTTP) request being sent to server using `--request` option. For example,
```
Comment (simo5) [inline USAGE.md]: using -> using the

File: USAGE.md [hunk 5]
```diff
@@ -417,8 +417,50 @@ and [test-truncating-of-kRSA-client-key-exchange.py](https://github.com/tomato42
...
+to perform a GET request for specific file, not the `/` object.
```
Comment (simo5) [inline USAGE.md]: for a specific file instead of the '/' object.

File: USAGE.md [hunk 6]
```diff
@@ -417,8 +417,50 @@ and [test-truncating-of-kRSA-client-key-exchange.py](https://github.com/tomato42
...
+
+`--cookie` option needs to be used if the server sends `cookie` extension in
```
Comment (simo5) [inline USAGE.md]: needs -> should/must

File: USAGE.md [hunk 7]
```diff
@@ -417,8 +417,50 @@ and [test-truncating-of-kRSA-client-key-exchange.py](https://github.com/tomato42
...
+HelloRetryRequest message.
+
+`--supported-groups` needs to be used when the server sends `supported_groups`
```
Comment (simo5) [inline USAGE.md]: needs -> should/must

File: USAGE.md [hunk 8]
```diff
@@ -417,8 +417,50 @@ and [test-truncating-of-kRSA-client-key-exchange.py](https://github.com/tomato42
...
+in EncryptedExtension during regular TLS 1.3 handshakes while
```
Comment (simo5) [inline USAGE.md]: regular -> a normal

File: USAGE.md [hunk 9]
```diff
@@ -417,8 +417,50 @@ and [test-truncating-of-kRSA-client-key-exchange.py](https://github.com/tomato42
...
+`--hrr-supported-groups` is to be used when the server does that for handshakes
```
Comment (simo5) [inline USAGE.md]: is to be -> should be
also what is "that" ?

File: USAGE.md [hunk 10]
```diff
@@ -417,8 +417,50 @@ and [test-truncating-of-kRSA-client-key-exchange.py](https://github.com/tomato42
...
+that force the server to send HelloRetryRequest message.
+
+In case the server does limit the size it advertises in its extension,
```
Comment (simo5) [inline USAGE.md]: does limit -> limits

File: USAGE.md [hunk 11]
```diff
@@ -417,8 +417,50 @@ and [test-truncating-of-kRSA-client-key-exchange.py](https://github.com/tomato42
...
+`--expect-size` can be used to set it. Note though, the value expected from
```
Comment (simo5) [inline USAGE.md]: then `--expect-size` can be used to set the expected limit.

File: USAGE.md [hunk 12]
```diff
@@ -417,8 +417,50 @@ and [test-truncating-of-kRSA-client-key-exchange.py](https://github.com/tomato42
...
+server in TLS 1.3 will be one byte larger than for TLS 1.2 and earlier
```
Comment (simo5) [inline USAGE.md]: than for -> than that for

File: USAGE.md [hunk 13]
```diff
@@ -417,8 +417,50 @@ and [test-truncating-of-kRSA-client-key-exchange.py](https://github.com/tomato42
...
+protocols as this is the expected behaviour of servers that support the biggest
+possible records and it makes the size of actual application data payload the
```
Comment (simo5) [inline USAGE.md]: full stop after "records" then start a new sentence.

**Comment by tomato42** [comment]
@simo5 fixed all nits, please re-review

**Comment by tomato42** [comment]
thanks for review!

**Pull request merged**